.github/workflows/build.yml

```diff
@@ -0,0 +1,116 @@
+# Thanks to this blog post: https://data-dive.com/multi-os-deployment-in-cloud-using-pyinstaller-and-github-actions/
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    # tags:
+    #   - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  # create-release:
+  #   permissions:
+  #     contents: write
+  #   runs-on: ubuntu-latest
+  #   outputs:
+  #     release_id: ${{ steps.create-release.outputs.id }}
+    
+  #   steps:
+  #     - uses: actions/checkout@v3
+  #     - uses: actions/setup-node@v3
+  #     - name: Create release
+  #       id: create-release
+  #       uses: actions/github-script@v6
+  #       with:
+  #         script: |
+  #           const { data } = await github.rest.repos.createRelease({
+  #             owner: context.repo.owner,
+  #             repo: context.repo.repo,
+  #             tag_name: context.ref,
+  #             name: 'Release ' + context.ref,
+  #             body: 'This is a test release to download and install Ref Studio',
+  #             draft: true,
+  #             prerelease: false
+  #           })
+  #           return data.id
+    
+  build-tauri:
+    # needs: create-release
+    permissions:
+      contents: write
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
Now we have an os matrix, you need to change the cache key for poetry `key: poetry-cache-${{ runner.os }}-${{ hashFiles('.github/actions/setup-python/action.yml') }}` (in `.github/actions/setup-python/action.yml`, line 15)